pandas/io/pickle.py

```diff
@@ -1,199 +0,0 @@
-    Load pickled pandas object (or any object) from file.
-
-    .. warning::
-
-       Loading pickled data received from untrusted sources can be
-       unsafe. See `here <https://docs.python.org/3/library/pickle.html>`__.
-
-    Parameters
-    ----------
-    filepath_or_buffer : str, path object or file-like object
-        File path, URL, or buffer where the pickled object will be loaded from.
-
-        .. versionchanged:: 1.0.0
-           Accept URL. URL is not limited to S3 and GCS.
-
-    compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None}, default 'infer'
-        If 'infer' and 'path_or_url' is path-like, then detect compression from
-        the following extensions: '.gz', '.bz2', '.zip', or '.xz' (otherwise no
-        compression) If 'infer' and 'path_or_url' is not path-like, then use
-        None (= no decompression).
-
-    Returns
-    -------
-    unpickled : same type as object stored in file
-
-    See Also
-    --------
-    DataFrame.to_pickle : Pickle (serialize) DataFrame object to file.
-    Series.to_pickle : Pickle (serialize) Series object to file.
-    read_hdf : Read HDF5 file into a DataFrame.
-    read_sql : Read SQL query or database table into a DataFrame.
-    read_parquet : Load a parquet object, returning a DataFrame.
-
-    Notes
-    -----
-    read_pickle is only guaranteed to be backwards compatible to pandas 0.20.3.
-
-    Examples
-    --------
-    >>> original_df = pd.DataFrame({"foo": range(5), "bar": range(5, 10)})
-    >>> original_df
-       foo  bar
-    0    0    5
-    1    1    6
-    2    2    7
-    3    3    8
-    4    4    9
-    >>> pd.to_pickle(original_df, "./dummy.pkl")
-
-    >>> unpickled_df = pd.read_pickle("./dummy.pkl")
-    >>> unpickled_df
-       foo  bar
-    0    0    5
-    1    1    6
-    2    2    7
-    3    3    8
-    4    4    9
-
-    >>> import os
-    >>> os.remove("./dummy.pkl")
-    """
-    fp_or_buf, _, compression, should_close = get_filepath_or_buffer(
-        filepath_or_buffer, compression=compression
-    )
-    if not isinstance(fp_or_buf, str) and compression == "infer":
-        compression = None
-    f, fh = get_handle(fp_or_buf, "rb", compression=compression, is_text=False)
-
-    # 1) try standard library Pickle
-    # 2) try pickle_compat (older pandas version) to handle subclass changes
-    # 3) try pickle_compat with latin-1 encoding upon a UnicodeDecodeError
-
-    try:
-        excs_to_catch = (AttributeError, ImportError, ModuleNotFoundError)
-        try:
-            with warnings.catch_warnings(record=True):
-                # We want to silence any warnings about, e.g. moved modules.
-                warnings.simplefilter("ignore", Warning)
-                return pickle.load(f)
-        except excs_to_catch:
-            # e.g.
-            #  "No module named 'pandas.core.sparse.series'"
-            #  "Can't get attribute '__nat_unpickle' on <module 'pandas._libs.tslib"
-            return pc.load(f, encoding=None)
```

Comment:
Wouldn't it be easier to just remove the calls to `pickle_compat` instead of removing all pickle functionality? It looks like it only uses that for legacy pickles which we don't care about, as we don't go through here anyway.

Reply:
I have not removed all pickle functionality - all __getstate__, __setstate__ and __reduce__ functionality is still present. I am happy to restore these tests if we can get the tests to work. Currently they are disabled in our build scripts and I spent this afternoon trying to get them working without much luck: mostly issues with dateutil and numpy - the tests were probably written for an older version of these.

Reply:
Sorry, I meant all pickle tests, and the `to_pickle` function, not the methods on the objects themselves. There is already a case to fix running the panda tests through build scripts https://jira.seequent.com/browse/LF-42195.